### DISCUSSION TITLE
Do I have to 'hide' the settings.json file that has my API_KEY and SECRET with amazon s3 upload plugin?

### DISCUSSION THREAD

**Discussion by jrhager84** [strapi/strapi#7825] [Legacy Topics]
I've consulted my documentation everywhere, and I can't seem to figure out how to obfuscate the private keys for upload into a repository. You can't use process.env variables in a JSON file, so I'm not sure if it's something I have to gitignore, or something that is even client-facing. Any help would be appreciated. Thanks!

**Comment by derrickmehaffy** [comment]
In the stable release you would be using the `./config/plugins.js` or one of the environment overrides `./config/env/${NODE_ENV}/plugins.js` in which these have support for our new environment parsing which documentation can be found here: https://strapi.io/documentation/v3.x/concepts/configurations.html#environment-variables

We no longer use .json files for these, but when we did back in the beta you could pass in `process.env` keys to it and they would be parsed. I'm guessing you were looking at the old beta docs, which that version is no longer supported.

**Reply by jrhager84** [reply]
It seems so. Why does it always default to the beta docs? Is there a setting I can flip to keep from erroneously consulting those docs?

**Reply by derrickmehaffy** [reply]
It's an issue with SEO that we are working to solve, there is a notice placed in the upper left when you are viewing the beta docs.
